sympy/sets/sets.py

```diff
@@ -911,6 +911,15 @@ def _union(self, other):
 
         See Set._union for docstring
         """
+        if len(other.args)>1:
```

Comment:
@asmeurer  Can you please review and suggest me if any further changes are needed?